tests/test_ldd.py

```diff
@@ -0,0 +1,250 @@
+        ]
+
+        for i, line in enumerate(lines):
+            dep = lddwrap.parse_line(line=line)
+
+            self.assertTrue(
+                equals(dep, expected_deps[i]), "Incorrect dependency read from the line {}, expected {}\ngot {}".format(
+                    line, expected_deps[i], dep))
+
+    def test_parse_wrong_lines(self):
+        lines = ["some wrong data which does not make sense", ""]
+
+        for line in lines:
+            self.assertRaises(Exception, lddwrap.parse_line, line)
+
+    def test_pwd(self):
+        path = pathlib.Path("/bin/pwd")
+        deps = lddwrap.get_dependencies(path=path)
+
+        expected_deps = [
+            lddwrap.Dependency(soname="linux-vdso.so.1", path=None, found=True, mem_address=""),
+            lddwrap.Dependency(
+                soname="libc.so.6", path=pathlib.Path("/lib/x86_64-linux-gnu/libc.so.6"), found=True, mem_address=""),
+            lddwrap.Dependency(
+                soname=None, path=pathlib.Path("/lib64/ld-linux-x86-64.so.2"), found=True, mem_address="")
+        ]
+
+        for i, dep in enumerate(deps):
+            self.assertTrue(
+                equals(dep, expected_deps[i]), "Incorrect dependency, expected {}\ngot {}".format(
+                    expected_deps[i], dep))
+
+    def test_dir(self):
+        path = pathlib.Path("/bin/dir")
+        deps = lddwrap.get_dependencies(path=path)
+
+        expected_deps = [
+            lddwrap.Dependency(soname="linux-vdso.so.1", path=None, found=True, mem_address=""),
+            lddwrap.Dependency(
+                soname="libselinux.so.1",
+                path=pathlib.Path("/lib/x86_64-linux-gnu/libselinux.so.1"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname="libc.so.6", path=pathlib.Path("/lib/x86_64-linux-gnu/libc.so.6"), found=True, mem_address=""),
+            lddwrap.Dependency(
+                soname="libpcre.so.3",
+                path=pathlib.Path("/lib/x86_64-linux-gnu/libpcre.so.3"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname="libdl.so.2", path=pathlib.Path("/lib/x86_64-linux-gnu/libdl.so.2"), found=True, mem_address=""),
+            lddwrap.Dependency(
+                soname=None, path=pathlib.Path("/lib64/ld-linux-x86-64.so.2"), found=True, mem_address=""),
+            lddwrap.Dependency(
+                soname="libpthread.so.0",
+                path=pathlib.Path("/lib/x86_64-linux-gnu/libpthread.so.0"),
+                found=True,
+                mem_address="")
+        ]
+
+        for i, dep in enumerate(deps):
+            expected_dep = expected_deps[i]
+            self.assertTrue(equals(dep, expected_dep))
+
+    def test_pwd_option_verbose(self):
+        path = pathlib.Path("/bin/pwd")
+        deps = lddwrap.get_dependencies(path=path, verbose=True)
```

Comment:
Comment D:

Since you tested quiet (non verbose) `get_dependencies` before, here it is enough to do

```
verbose_deps = get_deps(verbose=True)
quiet_deps = get_deps(ver=False)

for vd, qd in zup(ver_deps, q_deps):
  self.assertTrue(equals(..., ...), "message...")
```